### PR TITLE
Added Pinned option and its disability condition while editing Flow

### PR DESCRIPTION
**Pull request by enigmaticmahesh**
## Summary

1. Added an option to pin flow while editing the flow
2. Disable this option if the flow is pinned through organization settings.
3. Issue #2096

### CODE REVIEW

**Comment by enigmaticmahesh** [comment]
@mdshamoon I saw some merge conflicts in this PR, I think both the code should reside and it would resolve it.

**Review by kurund — CHANGES_REQUESTED** [review]
Can you please resolve the conflicts?

**Comment by enigmaticmahesh** [comment]
@kurund resolved the conflicts

**Review by mdshamoon — CHANGES_REQUESTED** [review]
@enigmaticmahesh can you please check the failing unit tests.

**Comment by enigmaticmahesh** [comment]
@mdshamoon fixed the unit test cases

File: src/containers/Flow/Flow.test.tsx [hunk 1]
```diff
@@ -53,13 +60,22 @@ it('should support keywords in a separate language', async () => {
 });
 
 it('should not allow special characters in keywords', async () => {
-  const { container, getByText } = render(flow({ params: {} }));
+  const { container, getByText, findByTestId } = render(flow({ params: {} }));
 
-  await waitFor(() => {});
-  fireEvent.change(container.querySelector('input[name="name"]'), {
+  const nameInput = await (await findByTestId('formLayout')).querySelector('input[name="name"]');
```
Comment (kurund) [inline src/containers/Flow/Flow.test.tsx:65]: Can we check a better way to achieve this? The code looks bit funky :)

**Review by kurund — CHANGES_REQUESTED** [review]

File: src/containers/Flow/Flow.test.tsx (hunk 1 above)
Comment (enigmaticmahesh) [reply src/containers/Flow/Flow.test.tsx:65]: @kurund I referred it directly from docs, so I used it. Ok, let me check for other ways.

Comment (kurund) [reply src/containers/Flow/Flow.test.tsx]: @enigmaticmahesh I was referring to await before await which you have removed now. So it seems fine.

**Pull request merged**